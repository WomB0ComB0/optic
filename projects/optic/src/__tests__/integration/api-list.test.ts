import {
  test,
  expect,
  describe,
  jest,
  beforeEach,
  afterEach,
} from '@jest/globals';
import { runOptic, setupWorkspace, normalizeWorkspace } from './integration';
jest.setTimeout(30000);

describe('optic api list', () => {
  let oldEnv: any;
  beforeEach(() => {
    oldEnv = { ...process.env };
    process.env.LOG_LEVEL = 'info';
    process.env.OPTIC_ENV = 'test';
  });

  afterEach(() => {
    process.env = { ...oldEnv };
  });

  test('lists all files in repo', async () => {
    process.env.OPTIC_TOKEN = 'something';

    const workspace = await setupWorkspace('api-list/many-files');

    const { combined, code } = await runOptic(workspace, 'api list');

    expect(code).toBe(0);
    expect(normalizeWorkspace(workspace, combined)).toMatchSnapshot();
  });
});
