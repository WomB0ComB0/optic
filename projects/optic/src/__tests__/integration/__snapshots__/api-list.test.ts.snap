// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optic api list lists all files in repo 1`] = `
"[1m[90mLooking for OpenAPI specs in directory $workspace$[39m[22m
example-api-v0.json [31m (untracked)[39m
spec.yml [31m (untracked)[39m
nested/speccopy2.yml [31m (untracked)[39m

[34m[1mRun optic api add to add untracked apis[22m[39m
"
`;
